final/client/src/pages/launch.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Fragment } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+
+import { LAUNCH_TILE_DATA } from './launches';
+import { Loading, Header, LaunchDetail } from '../components';
+import { ActionButton } from '../containers';
+import { RouteComponentProps } from '@reach/router';
+import { QueryResult } from '@apollo/react-common';
+import * as LaunchDetailsTypes from './__generated__/LaunchDetails';
+
+export const GET_LAUNCH_DETAILS = gql`
+  query LaunchDetails($launchId: ID!) {
+    launch(id: $launchId) {
+      isInCart @client
+      site
+      rocket {
+        type
+      }
+      ...LaunchTile
+    }
+  }
+  ${LAUNCH_TILE_DATA}
+`;
+
+interface LaunchProps extends RouteComponentProps {
+  launchId?: any;
+}
+
+const Launch: React.FC<LaunchProps> = ({ launchId }) => {
+  const { 
+    data, 
+    loading, 
+    error 
+  }: QueryResult<
+    LaunchDetailsTypes.LaunchDetails, 
+    LaunchDetailsTypes.LaunchDetailsVariables
+  > = useQuery(
```

Comment:
No need to specify `QueryResult`. You can type this as:

```ts
const { data, loading, error } = useQuery<
    LaunchDetailsTypes.LaunchDetails,
    LaunchDetailsTypes.LaunchDetailsVariables
>(GET_LAUNCH_DETAILS, { variables: { launchId } });
```

Reply:
Fixed!